### PR TITLE
Suppress error messages when using ConfigFile::get_value and a default is given

### LINKED ISSUE #8097: ConfigFile::get_value error reporting
I am ConfigFile::get_value and it takes a default value. The problem is it reports an error if it decides to use this value, which is what I would expect it to do!
So it is working as one would expect yet throwing out errors or warnings.
I keep seeing errors as I run my game yet it is working perfectly as expected.

Can we stop this method from reporting the false errors?

### PR DESCRIPTION
**Pull request by bojidar-bg**
Fixes #8097

Not sure if I should remove the if on lines 86-89 (`if(!p_default`)) thought..

### CODE REVIEW

**Comment by bojidar-bg** [comment]
(I won't make a 2.1 PR for this, since it should cherrypick easily)

**Comment by bojidar-bg** [comment]
Oops, clang-format failed... :smile:

File: core/io/config_file.cpp [hunk 1]
```diff
@@ -82,8 +82,13 @@ void ConfigFile::set_value(const String &p_section, const String &p_key, const V
 }
 Variant ConfigFile::get_value(const String &p_section, const String &p_key, Variant p_default) const {
 
-	ERR_FAIL_COND_V(!values.has(p_section), p_default);
-	ERR_FAIL_COND_V(!values[p_section].has(p_key), p_default);
+	if (!values.has(p_section) || !values[p_section].has(p_key)) {
+		if (!p_default) {
```
Comment (karroffel) [inline core/io/config_file.cpp]: Wouldn't this print an error message when the default value was `Variant(false)`?

Comment (bojidar-bg) [inline core/io/config_file.cpp]: Good point, I guess I should go back and fix it...

**Review by karroffel — APPROVED** [review]

**Comment by akien-mga** [comment]
I'm unsure about this change. Now it would become pretty difficult to actually notice when you are trying to get a non-existing value, as it would just silently return the default. And since it's a builtin function that's not something where users could put a debug statement to actually notice it if they care.

**Comment by bojidar-bg** [comment]
@akien-mga If you _are_ giving a default value, we should expect you to know how are you going to use it. If you don't give a default (or pass null), it would still throw errors.

**Pull request merged**